b2share/modules/upgrade/cli.py

```diff
@@ -0,0 +1,123 @@
+@click.option('-v', '--verbose', is_flag=True, default=False)
+def run(verbose):
+    """Upgrade the database, activate alembic and reindex the records."""
+    alembic = current_app.extensions['invenio-db'].alembic
+    new_migration = Migration()
+    new_migration.version = __version__
+    new_migration.date = datetime.utcnow()
+    migration_data = {'elasticsearch': False,
+                      'alembic_version': False,
+                      'alembic_upgrade': False}
+
+    if (db.engine.dialect.has_table(db.engine, 'transaction') and
+            not alembic.migration_context._has_version_table()):
+
+        # Make sure alembic version table exists in the db
+        alembic.migration_context._ensure_version_table()
+
+        # alembic_version state of B2Share v2.0.1
+        heads_v2_0_1 = ['dbdbc1b19cf2', '1ba76da94103',
+                        '12a88921ada2', '97bbc733896c',
+                        '2f63be7b7572', 'e655021de0de',
+                        '999c62899c20', 'fb99eeaec4ac',
+                        '35d7d8958395']  # '35c1075e6360'
+
+        # Populate it as if alembic upgrade heads had ran
+        for head in heads_v2_0_1:
+            alembic.migration_context.stamp(
+                alembic.script_directory, head)
+        migration_data['alembic_version'] = True
+
+    # Upgrade if there are still outdated models
+    if verbose:
+        click.secho('Upgrading...', fg='green')
+    try:
+        alembic.upgrade()
+        click.secho('Done', fg='green')
+        migration_data['alembic_upgrade'] = True
+    except Exception as e:
+        pass
+    with current_app.app_context():
+        db.session.add(new_migration)
+        db.session.commit()
+
+    call(['psql', '-U', 'postgres', '-d', 'b2sharedb',
+          '-a', '-f',
+          'b2share/modules/upgrade/alembic_renaming_script.sql'])
+
+    # Elasticsearch reindexing
+    runner = CliRunner()
+    script_info = ScriptInfo(create_app=lambda info: current_app)
+    result_destroy = runner.invoke(index_cmd,
+                                   ['destroy', '--yes-i-know', '--force'],
+                                   obj=script_info)
+    migration_data['destroy'] = result_destroy.output
+    result_init = runner.invoke(index_cmd,
+                                ['init'],
+                                obj=script_info)
+    migration_data['init'] = result_init.output
+    result_reindex = runner.invoke(index_cmd,
+                                   ['reindex', '--yes-i-know'],
+                                   obj=script_info)
+    migration_data['reindex'] = result_reindex.output
+    result_run = runner.invoke(index_cmd,
+                               ['run'],
+                               obj=script_info)
+    migration_data['run'] = result_run.output
+    migration_data['elasticsearch'] = True
+
+    # Save the migration details
+    new_migration.data = migration_data
+    with current_app.app_context():
+        db.session.add(new_migration)
```

Comment:
This should be done if any exception happen during the upgrade. TODO: add a `try except`.